### ISSUE TITLE
Feature Request: Add Branch Option for Remote Repository Packing

### ISSUE THREAD

**Issue by tranquochuy645** [yamadashy/repomix#195]
Hello @yamadashy ,

I'm new to this project but I'm willing to help creating a PR. I think it would be beneficial to add an option to select a specific commit ID or branch when packing a remote repository.

Current Behavior:
When packing a remote repository, the process uses the default branch.

Proposed Feature:
Add a branch option to the packing process, allowing users to select a specific commit ID or branch to pack.

Potential Implementation:

```
repomix --remote https://github.com/yamadashy/repomix --branch master
```

**Comment by yamadashy** [comment]
Hi, @tranquochuy645 !
Wow, thank you for the detailed feature proposal and implementation! 

I've quickly reviewed the code and it looks great! Just one suggestion: since this is paired with the `--remote` option, could you rename the options to `--remote-branch` for better consistency? 

I'll review the code more thoroughly later. Really appreciate your contribution!

**Comment by tranquochuy645** [comment]
Hey @yamadashy,

Thanks for the feedback! I’ve updated the PR to rename the option to --remote-branch for consistency with the --remote option, as you suggested. 

Let me know if anything else needs adjustment  👍

**Comment by yamadashy** [comment]
Hi, @tranquochuy645 
I've released v0.2.6 which includes your `--remote-branch` feature.
Thanks again for implementing this!

You can check out the release notes here:
https://github.com/yamadashy/repomix/releases/tag/v0.2.6

**Comment by darrelladjei** [comment]
Hey guys, it seems the commit option isn't supported? 
I tried with `npx repomix --remote https://github.com/cubiq/ComfyUI_IPAdapter_plus --remote-branch b188a6c`. 
Which should point to https://github.com/cubiq/ComfyUI_IPAdapter_plus/tree/b188a6c 
Any help much appreciated!

**Comment by tranquochuy645** [comment]
Hello @darrelladjei ,

Thank you for reporting this issue.
Apologies for the oversight. I didn’t verify whether it also works with a commit hash. It turns out that `git clone -b` only supports branch names and tags.

I'm currently working on another PR to fix this issue!

**Comment by darrelladjei** [comment]
Hey @tranquochuy645 thanks for addressing, let me know if I should also take a look! Haven't contributed before but willing to help get this over the line! 
Any idea of time lines for your PR or even a way for me to hack this myself? 
Thanks in advance.

**Comment by tranquochuy645** [comment]
> Hey @tranquochuy645 thanks for addressing, let me know if I should also take a look! Haven't contributed before but willing to help get this over the line! Any idea of time lines for your PR or even a way for me to hack this myself? Thanks in advance.

Thanks for your interest and willingness to help!

I have created a draft PR https://github.com/yamadashy/repomix/pull/212 .  It works perfectly on my machine, but it still needs additional unit tests, and I’m not entirely sure how to write those properly.
If you’d like, feel free to try out the PR yourself and let me know how it works for you. Any feedback or help with writing the tests would be greatly appreciated!

**Comment by darrelladjei** [comment]
Nice thanks, thanks for that! It wasn't working for my use case when i ran it locally (using the Comfy_UI example from a few messages above). 
I had a quick fix i just wrote tests for so made PR for it [here](https://github.com/yamadashy/repomix/pull/213).
Let me know what you think too!

**Comment by tranquochuy645** [comment]
Hey @darrelladjei,

Thanks for your feedback! I’ve tested your use case, and I can confirm that my implementation doesn't work with the short commit hash (b188a6c), but it does work with the full hash (b188a6cb39b512a9c6da7235b880af42c78ccd0d).

This issue arises because git fetch doesn't support short commit hashes. However, I believe this limitation is acceptable given the circumstances.

The goal of my implementation is to avoid fetching any git history, and the optimal approach for this has been discussed in a previous Stack Overflow thread here: [How to clone git repository with a specific revision/changeset](https://stackoverflow.com/questions/3489173/how-to-clone-git-repository-with-specific-revision-changeset).

Regarding your PR, I don't think git checkout will work for commits on branches other than the default branch unless you perform a git fetch first. That said, the tests are great, and I plan to adapt some of them.

Thanks again!

**Comment by tranquochuy645** [comment]
This should be resolved with https://github.com/yamadashy/repomix/pull/212

**Comment by yamadashy** [comment]
Hi, @darrelladjei and @tranquochuy645 !

Just wanted to let you know that the fix for the commit SHA issue and the other related improvements are now released in v0.2.11!

https://github.com/yamadashy/repomix/releases/tag/v0.2.11

Thanks to @tranquochuy645's contribution, the `--remote-branch` option now supports specific commit hashes. Also, thanks to @darrelladjei for taking the initiative to test the commit SHA functionality and for opening a PR with the test cases!

Thank you both for your feedback and contributions! They've been invaluable in making Repomix better.

Let me know if you have any further questions or suggestions!
